api/envoy/config/filter/http/compressor/v2/compressor.proto

```diff
@@ -45,4 +50,25 @@ message Compressor {
   // Runtime flag that controls whether the filter is enabled or not. If set to false, the
   // filter will operate as a pass-through filter. If not specified, defaults to enabled.
   api.v2.core.RuntimeFeatureFlag runtime_enabled = 5;
+
+  // A compressor library to use for compression. By default
+  // :ref:`envoy.filters.http.compressor.gzip<envoy_api_msg_config.filter.http.compressor.gzip.v2.Gzip>`
+  // is used.
+  //
+  // This field is ignored when used in the context of the deprecated "envoy.filters.http.gzip"
+  // filter.
+  CompressorLibrary compressor_library = 6;
```

Comment:
I'm not sure it makes sense to have a default here. Can we require this and then make the gzip config explicit for clarity?

Reply:
Can we make it required via annotation?

Reply:
Hm... This message is used also in the deprecated `envoy.filters.http.gzip` filter. I guess there are two options:

1. revert #10306 completely or partially (drop the `compressor` field from the `Gzip` message);
2. duplicate `compressor.proto` for `envoy.filters.http.gzip`.